src/bentoml/_internal/store.py

```diff
@@ -170,6 +170,9 @@ def register(self, tag: t.Union[str, Tag]):
         self._fs.makedirs(item_path)
         try:
             yield self._fs.getsyspath(item_path)
+        except Exception:
+            self._fs.removetree(item_path)
+            raise
         finally:
```

Comment:
I didn't want you to change it back to a finally, I agree this should not execute when there's an exception. Let's just dedent the block and remove the finally?